packages/components/src/list/style.scss

```diff
@@ -26,6 +26,46 @@ a.woocommerce-list__item {
 			inset 0 0 0 2px $studio-white;
 	}
 
+	&:focus-visible {
+		box-shadow: none;
+	}
+
+	// transitions
+	&.enter {
+		opacity: 0;
+		max-height: 0;
+		transform: translateX( 50% );
+	}
+
+	&.enter-active {
+		opacity: 1;
+		max-height: 100vh;
+		transform: translateX( 0% );
+		transition: opacity 500ms, transform 500ms, max-height 500ms;
+	}
+
+	&.exit {
+		opacity: 1;
+		max-height: 100vh;
+		transform: translateX( 0% );
+	}
+
+	&.exit-active {
+		opacity: 0;
+		max-height: 0;
+		transform: translateX( 50% );
+		transition: opacity 500ms, transform 500ms, max-height 500ms;
+	}
+
+	&.transitions-disabled {
+		&.enter-active,
+		&.exit,
+		&.exit-active {
+			transition: none;
+			transform: none;
```

Comment:
Hmm instead of overwriting the transition properties when it is disabled, would it be worth to wrap the transitions instead with either `:not(.transition-disabled)` or something specific to the transition `.list-slide-right`?

Reply:
I think I did it this way to not impact the existing component, but afaict transitions on the old one may have been broken anyway because when react-transition-group was updated the format of the expected classes for transitions changed. I'll explore your suggestions here 👍

Reply:
Addressed here: https://github.com/woocommerce/woocommerce-admin/pull/6787/commits/5d04cb4baa82c80294c58ba6a591d27bdca3e319